File: python/ex-3.py
file = open("words.txt", "r")
data = file.read()
words = data.split(' ')
st = ''
for i in words:
    st = st + i 
print('sr' ,st)
wordToReplace = input("Enter word to replace ")

"""
3- Replace the word into file and count.

"""

replacementWord = input("Enter replacement word ")
count = data.count(wordToReplace)
print('Found {} words in file '.format(count))
if count > 0:
    data = data.replace(wordToReplace, replacementWord)
    #file.truncate(0)
    print(data)
    with open("words.txt", "w") as w:
        w.write(data)
    #file.write(data)

    print('Replacemt done') 
else:
    print("No word found")

File: go/ex-3.go
package main

import (
	"fmt"
	"os"
	"strings"
)

/**
Exercise - 3

3- Replace the word into file and count.

**/

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func main() {
	dat, err := os.ReadFile("words.txt")
	check(err)
	content := string(dat)

	println("Enter word to replace: ")
	var findWord string
	fmt.Scanln(&findWord)

	println("Enter replacement word: ")
	var replacementWord string
	fmt.Scanln(&replacementWord)
	output := ""
	words := strings.Split(content, " ")
	for _, word := range words {
		if strings.TrimRight(word, "\r\n") == findWord {
			os.WriteFile("words-out", []byte(output), 0644)
			println("Replacemt done")
			os.Exit(0)
		}
	}
	println("Word not found")

}
